crypto/fips1402/src/test/java/org/keycloak/crypto/fips/test/FIPS1402SecureRandomTest.java

```diff
@@ -0,0 +1,52 @@
+package org.keycloak.crypto.fips.test;
+
+import java.security.SecureRandom;
+
+import org.jboss.logging.Logger;
+import org.junit.Assume;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.keycloak.common.crypto.CryptoIntegration;
+import org.keycloak.common.util.Environment;
+import org.keycloak.rule.CryptoInitRule;
+
+/**
+ * @author <a href="mailto:mposolda@redhat.com">Marek Posolda</a>
+ */
+public class FIPS1402SecureRandomTest {
+
+    @ClassRule
+    public static CryptoInitRule cryptoInitRule = new CryptoInitRule();
+
+    @Before
+    public void before() {
+        // Run this test just if java is in FIPS mode
+        Assume.assumeTrue("Java is not in FIPS mode. Skipping the test.", Environment.isJavaInFipsMode());
+    }
+
+    protected static final Logger logger = Logger.getLogger(FIPS1402SecureRandomTest.class);
+
+    @Test
+    public void testSecureRandom() throws Exception {
+        logger.info(CryptoIntegration.dumpJavaSecurityProviders());
+
+        SecureRandom sc1 = new SecureRandom();
+        logger.infof(dumpSecureRandom("new SecureRandom()", sc1));
+
+        SecureRandom sc2 = SecureRandom.getInstance("DEFAULT", "BCFIPS");
+        logger.infof(dumpSecureRandom("SecureRandom.getInstance(\"DEFAULT\", \"BCFIPS\")", sc2));
+
+        SecureRandom sc3 = SecureRandom.getInstance("SHA1PRNG");
+        logger.infof(dumpSecureRandom("SecureRandom.getInstance(\"SHA1PRNG\")", sc3));
```

Comment:
Shouldn't this be some asserts instead of logging?

Reply:
Added some asserts. Same as for the comment above.